tests/nest_tests/test_integrate_odes.py

```diff
@@ -232,3 +230,50 @@ def test_integrate_odes_params2(self):
         generate_target(input_path=fname, target_platform="NONE", logging_level="DEBUG")
 
         assert len(Logger.get_all_messages_of_level_and_or_node("integrate_odes_test", LoggingLevel.ERROR)) == 2
+
+    def test_integrate_odes_higher_order(self):
+        r"""
+        Tests for higher-order ODEs of the form F(x'',x',x)=0, integrate_odes(x) integrates the full dynamics of x.
+        """
+        resolution = 0.1
+        simtime = 15.
+        nest.set_verbosity("M_ALL")
+        nest.ResetKernel()
+        nest.SetKernelStatus({"resolution": resolution})
+        try:
+            nest.Install("nestmlmodule")
+        except Exception:
+            # ResetKernel() does not unload modules for NEST Simulator < v3.7; ignore exception if module is already loaded on earlier versions
+            pass
+
+        n = nest.Create("alpha_function_2nd_order_ode_neuron_nestml")
+        sgX = nest.Create("spike_generator", params={"spike_times": [10.]})
+        nest.Connect(sgX, n, syn_spec={"weight": 1., "delay": resolution})
+
+        mm = nest.Create("multimeter", params={"interval": resolution, "record_from": ["x", "y"]})
+        nest.Connect(mm, n)
+
+        nest.Simulate(simtime)
+        x = mm.get()["events"]["x"]
+        y = mm.get()["events"]["y"]
+        times = mm.get()["events"]["times"]
+        print(x[-1], y[-1])
+
+        if TEST_PLOTS:
+            fig, ax = plt.subplots(nrows=2)
+            ax1, ax2 = ax
+
+            ax2.plot(times, x, label="x")
+            ax1.plot(times, y, label="y", alpha=.7, linestyle=":")
+
+            for _ax in ax:
+                _ax.grid(which="major", axis="both")
+                _ax.grid(which="minor", axis="x", linestyle=":", alpha=.4)
+                _ax.set_xlim(0., simtime)
+                _ax.legend()
+
+            fig.savefig("/tmp/test_integrate_odes_higher_order.png", dpi=300)
+
+        # verify
+        np.testing.assert_allclose(x[-1], 0.10737970490959549)
```

Comment:
How do we know these values are correct (are they from just running the NESTML model once, or from an independent implementation of the ODEs)?

Reply:
These values are from repeated simulations of the NESTML-generated code, equivalent to providing `integrate_odes(x',x)` in the NESTML model. I have changed the test now to include calls to both `integrate_odes` and compare their output.